CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 3.1.0)
+
+project(TwitchWindow VERSION 1.0.0 LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+find_package(Qt5 COMPONENTS Core Widgets Gui OpenGL REQUIRED)
+find_package(OpenCV REQUIRED)
+find_package(LibAV REQUIRED)
+
+set(SRC_DIR ${PROJECT_SOURCE_DIR}/src/)
+
+set(SRC_COMPONENTS
+    ${SRC_DIR}Components/DataType.h
+    ${SRC_DIR}Components/Application.h
+    ${SRC_DIR}Components/Application.cpp
+    ${SRC_DIR}Components/Component.h
+    ${SRC_DIR}Components/Capture.h
+    ${SRC_DIR}Components/Capture.cpp
+    ${SRC_DIR}Components/Controller.h
+    ${SRC_DIR}Components/Controller.cpp
+    ${SRC_DIR}Components/Encoder.h
+    ${SRC_DIR}Components/Encoder.cpp
+    ${SRC_DIR}Components/Decoder.h
+    ${SRC_DIR}Components/Decoder.cpp
+    ${SRC_DIR}Components/Render.h
+    ${SRC_DIR}Components/Render.cpp
+    ${SRC_DIR}Components/Router.h
+    ${SRC_DIR}Components/Router.cpp
+    )
+
+include_directories(${SRC_DIR})
+include_directories(${LIBAV_INCLUDE_DIR})
+include_directories(${OpenCV_INCLUDE_DIRS})
+
+add_Definitions("-D_XKEYCHECK_H")
+add_Definitions("-D_USE_MATH_DEFINES")
+
+add_executable(TwitchWindow
+    ${SRC_DIR}main.cpp
+
+    ${SRC_DIR}TwitchWindow.ui
+    ${SRC_DIR}TwitchWindow.qrc
+
+    ${SRC_DIR}TwitchWindow.h
+    ${SRC_DIR}TwitchWindow.cpp
+
+    ${SRC_DIR}MyGLWidget.h
+    ${SRC_DIR}MyGLWidget.cpp
+
+    ${SRC_COMPONENTS}
+)
+
+source_group("Components" FILES ${SRC_COMPONENTS})
+
+target_link_libraries(TwitchWindow 
+    ${OpenCV_LIBRARIES}
+    ${LIBAV_LIBRARIES}
+    wsock32.lib
+    OpenGL32.lib
+    )
+
+target_link_libraries(TwitchWindow 
+    Qt5::Core
+    Qt5::Widgets
+    Qt5::Gui
+    Qt5::OpenGL
+    )
+
+find_path(OpenCV_BIN 
```

Comment:
Windows specific build options should be under an if block that is active under windows only.

Reply:
Same goes to all dll below.